be/src/formats/parquet/parquet_file_writer.h

```diff
@@ -107,6 +107,14 @@ class ParquetFileWriter final : public FileWriter {
     const std::function<void()> _rollback_action;
     PriorityThreadPool* _executors = nullptr;
     RuntimeState* _runtime_state = nullptr;
+
+    struct ExecutionState {
+        std::mutex mu;
+        std::condition_variable cv;
+        bool has_unfinished_task = false;
+    };
```

Comment:
count down latch is what you needs actually, you can find it in be/src/util/count_down_latch.h

Reply:
After some digging, I think `CountDownLatch` does not quite fit my needs. What I do want is a binary state that flips, not a counter.

Reply:
I think `has_unfinished_task` means that the number of unfinished tasks is greater than 0. Actually I don't got you why the count down can't fulfill binary flips.